packages/vercel-ai-provider/src/edgedb-prepare-tools.ts

```diff
@@ -0,0 +1,152 @@
+import type {
+  JSONSchema7,
+  LanguageModelV1,
+  LanguageModelV1CallWarning,
+} from "@ai-sdk/provider";
+import {
+  type EdgeDBChatModelId,
+  isAnthropicModel,
+  isOpenAIModel,
+} from "./edgedb-chat-settings";
+
+interface OpenAILikeTool {
+  type: "function";
+  function: {
+    name: string;
+    description: string | undefined;
+    parameters: unknown;
+  };
+}
+
+interface AnthropicTool {
+  name: string;
+  description: string | undefined;
+  input_schema: JSONSchema7;
+}
+
+export function prepareTools(
+  mode: Parameters<LanguageModelV1["doGenerate"]>[0]["mode"] & {
+    type: "regular";
+  },
+  model: EdgeDBChatModelId,
+) {
+  const isOpenAI = isOpenAIModel(model);
+  const isAnthropic = isAnthropicModel(model);
+
+  // when the tools array is empty, change it to undefined to prevent errors:
+  const tools = mode.tools?.length ? mode.tools : undefined;
+  const toolWarnings: LanguageModelV1CallWarning[] = [];
+
+  if (tools == null) {
+    return { tools: undefined, tool_choice: undefined, toolWarnings };
```

Comment:
Nit: do we need to set this explicitly to `undefined` or can we just return:

```suggestion
    return { toolWarnings };
```

Reply:
I didnt really try, all other providers return with undefined

Reply:
We shouldn't block on this, it's not _wrong_, but as I mentioned earlier, I personally find explicit use of `undefined` to be some kind of smell.